planner/core/task.go

```diff
@@ -574,7 +574,6 @@ func (p *PhysicalMergeJoin) attach2Task(tasks ...task) task {
 	lTask := finishCopTask(p.ctx, tasks[0].copy())
 	rTask := finishCopTask(p.ctx, tasks[1].copy())
 	p.SetChildren(lTask.plan(), rTask.plan())
-	p.schema = BuildPhysicalJoinSchema(p.JoinType, p)
```

Comment:
TODO: I didn't know why we need this

Reply:
Because we have already pruned extra columns in logical optimization here, https://github.com/pingcap/tidb/blob/d1aa2f5b09d5763c53838310318252b574de154e/planner/core/rule_column_pruning.go#L327

This line rebuilds the schema of Join and reverts the effect of what logical optimization did before. So it should be deleted.

Reply:
And this [line](https://github.com/pingcap/tidb/blob/6d00e2169bda543797aa3136cfcb5332911cd64d/planner/implementation/join.go#L67) in cascades planner should be deleted either, for the same reason